### PR TITLE
Add script to deprovision an OSP deployment

### PR DESCRIPTION
**Pull request by EmilienM**
This script will make sure we unregister RHSM when deploying on RHEL,
remove the server from Nova and then remove the config files from
ci-configs.

### CODE REVIEW

File: deprovision.sh [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+CLUSTER_NAME=$1
+SERVER_DELETE_CMD="openstack server delete --wait $CLUSTER_NAME"
+
+if [ ! -f "$ROOT_DIR/configs/$CLUSTER_NAME.yaml" ]; then
+    echo "ERROR: No config was found for $CLUSTER_NAME in $ROOT_DIR/configs"
+    exit 1
+fi
+
+# Sanity check for CI jobs and locally
+if [ -n "IS_CI" ]; then
```
Comment (MaysaMacedo) [inline deprovision.sh:71]: this probably should be: `$IS_CI`

File: deprovision.sh [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    for i in VEXXHOST_SHIFTSTACK_BM_CI_PASSWORD VEXXHOST_SHIFTSTACK_BM_CI_SSH_PRIVATE_KEY VEXXHOST_ENDPOINT; do
+       if [ -z "$i" ]; then
+           echo "ERROR:$ $i is not set and is required when this script runs in CI"
```
Comment (MaysaMacedo) [inline deprovision.sh:74]: Is this script meant to be run by CI?

Comment (EmilienM) [reply deprovision.sh:74]: at some point, our environments will be provisioned & unprovisioned dynamically.

**Comment by EmilienM** [comment]
this was tested and is ready for review.

**Pull request closed without merging**